pkg/controller/upgradeconfig/cluster_upgrader.go

```diff
@@ -0,0 +1,737 @@
+				continue
+			} else {
+				return false, err
+			}
+		}
+		if sub.Spec.Channel != item.Channel {
+			sub.Spec.Channel = item.Channel
+			err = c.Update(context.TODO(), sub)
+			if err != nil {
+				return false, err
+			}
+		}
+	}
+
+	return true, nil
+}
+
+// PostUpgradeVerification verify all replicasets are at expected counts and all daemonsets are at expected counts
+func PostUpgradeVerification(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+	replicaSetList := &appsv1.ReplicaSetList{}
+	err := c.List(context.TODO(), replicaSetList)
+	if err != nil {
+		return false, err
+	}
+	readyRs := 0
+	totalRs := 0
+	for _, replica := range replicaSetList.Items {
+		if strings.HasPrefix(replica.Namespace, "default") ||
+			strings.HasPrefix(replica.Namespace, "kube") ||
+			strings.HasPrefix(replica.Namespace, "openshift") {
+			totalRs = totalRs + 1
+			if replica.Status.ReadyReplicas == replica.Status.Replicas {
+				readyRs = readyRs + 1
+			}
+
+		}
+	}
+
+	if totalRs != readyRs {
+		reqLogger.Info(fmt.Sprintf("not all replicaset are ready:expected number :%v , ready number %v", len(replicaSetList.Items), readyRs))
+		return false, nil
+	}
+
+	dsList := &appsv1.DaemonSetList{}
+	err = c.List(context.TODO(), dsList)
+	if err != nil {
+		return false, err
+	}
+	readyDS := 0
+	totalDS := 0
+	for _, ds := range dsList.Items {
+		if strings.HasPrefix(ds.Namespace, "default") ||
+			strings.HasPrefix(ds.Namespace, "kube") ||
+			strings.HasPrefix(ds.Namespace, "openshift") {
+			totalDS = totalDS + 1
+			if ds.Status.DesiredNumberScheduled == ds.Status.NumberReady {
+				readyDS = readyDS + 1
+			}
+		}
+	}
+	if len(dsList.Items) != readyDS {
+		reqLogger.Info(fmt.Sprintf("not all daemonset are ready:expected number :%v , ready number %v", len(dsList.Items), readyDS))
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// TODO
+func RemoveMaintWindow(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+
+	return true, nil
+}
+
+// This perform cluster health check after upgrade
+func PostClusterHealthCheck(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+
+	return performClusterHealthCheck(c)
+}
+
+// Check whether nodes are upgraded or not
+func NodesUpgraded(c client.Client, nodeType string, reqLogger logr.Logger) (bool, error) {
+	configPool := &machineconfigapi.MachineConfigPool{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: nodeType}, configPool)
+	if err != nil {
+		return false, nil
+	}
+	//TODO send timeout alert if wait timeout
+	if configPool.Status.MachineCount != configPool.Status.UpdatedMachineCount {
+		errMsg := fmt.Sprintf("not all %s are upgraded, upgraded: %v, totall: %v", nodeType, configPool.Status.UpdatedMachineCount, configPool.Status.MachineCount)
+		reqLogger.Info(errMsg)
+		return false, fmt.Errorf(errMsg)
```

Comment:
This results in some pretty spammy error logging [0] during the node upgrades - but it's not really an "error" that some nodes still aren't yet upgraded. Thoughts on maybe just not returning an error at all here? The existing info logging is enough.

```suggestion
		return false, nil
```

[0] {"level":"error","ts":1591072217.1054864,"logger":"controller_upgradeconfig","msg":"error when AllWorkerNodesUpgraded","Request.Namespace":"","Request.Name":"example-upgrade-config","error":"no
t all worker are upgraded, upgraded: 6, totall: 8","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/mbargenq/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/o
penshift/managed-upgrade-operator/pkg/controller/upgradeconfig.ClusterUpgrader.UpgradeCluster\n\tmanaged-upgrade-operator/pkg/controller/upgradeconfig/cluster_upgrader.go:464\ngithub.com/openshift/managed-upgrade-operator/pkg/controller/upgradeconfig.(*ReconcileUpgradeConfig).Reconcile\n\tmanaged-upgrade-operator/pkg/controller/upgradeconfig/upgradeconfig_controller.go:143\nsigs.k8s.
io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/mbargenq/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:256\nsigs.k
8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/mbargenq/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:232\n
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/home/mbargenq/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:211\nk8s.io/
apimachinery/pkg/util/wait.JitterUntil.func1\n\t/home/mbargenq/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.Jit
terUntil\n\t/home/mbargenq/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/home/mbargenq/go/pkg/mod/k8s.
io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88"}

Reply:
+1 we want our logging spot on for this